src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -375,7 +375,7 @@ private ImmutableArray<ISymbol> GetSymbolsForExpressionOrStatementContext()
 
                 if (symbol.IsExtensionMethod() &&
                     !Equals(enclosingNamedType, symbol.ContainingType) &&
-                    !outerTypes.Any(outerType => outerType.Equals(symbol.ContainingType)))
+                    !outerTypes.Contains(symbol.ContainingType))
```

Comment:
Maybe there is some weird equality semantics for INamedTypeSymbol implementations that I'm not aware of?

Reply:
Nope. This is right